Utilities/Scripts/SlicerWizard/ExtensionProject.py

```diff
@@ -145,12 +147,46 @@ def project(self, value):
     raise EOFError("could not find project")
 
   #---------------------------------------------------------------------------
-  def getValue(self, name, default=None):
+  def substituteVariableReferences(self, text):
+    """Return a copy of ``text`` where all valid '``${var}``' occurrences
+    have been replaced.
+
+    :param text: A text with zero or more variable references.
+    :type text: :class:`basestring`
+    """
+    variableNames = self._referencedVariables.findall(text)
+    if len(variableNames) == 0:
+      return text
+
+    prefinedVariables = {}
+    try:
+       prefinedVariables[u"PROJECT_NAME"] = self.project
+    except EOFError:
+      pass
+
+    for name in prefinedVariables.keys():
```

Comment:
I wonder if this shouldn't handle nested references... I would have instead written the loop to find one reference, replace it, then start over, until no references remain. (As an optimization, you can pre-fill a cache of known variables with supported built-ins, i.e. `PROJECT_NAME`, check for hits, and add to the dict with `getValue` as needed.)

You would need a little clever logic to deal with unknown substitutions, though, assuming you want to leave them unexpanded (or you could expand them to `name-NOTFOUND`).